splipy/SplineObject.py

```diff
@@ -438,7 +439,111 @@ def raise_order(self, *raises):
             raise ValueError("Cannot lower order using raise_order")
         if all(r == 0 for r in raises):
             return
+        
+        if any(abs(b.knots[0] - b.knots[b.order-1]) > state.knot_tolerance or b.periodic > -1 for b in self.bases):
+            self.raise_order_implicit(*raises)
+            return
+        
+        new_bases = [b.raise_order(r) for b, r in zip(self.bases, raises)]
+
+        def raise_order_1D(n, k, T, P, m, periodic):
+            """ Implementation of method in "Efficient Degree Elevation and Knot Insertion
+                for B-spline Curves using Derivatives" by Qi-Xing Huang a Shi-Min Hu, Ralph R Martin. Only the case of open knot vector is fully implemented
+            :param int n: (n+1) is the number of initial basis functions
+            :param int k: spline order
+            :param T: knot vector
+            :param P: weighted NURBS coefficients
+            :param int m: number of degree elevations
+            :param int periodic: Number of continuous derivatives at start and end. -1 is not periodic, 0 is continuous, etc.
+            :return Q: new control points
+            """
+
+            u = np.unique(T[k-1:-k+1])
+            S = u.size - 1
+            d = P.shape[0] # dimension of spline
+
+            # Find multiplicity of the knot vector T
```

Comment:
See `BSplineBasis.continuity(knot)` which does this exact thing

Reply:
```  python
b = BSplineBasis(k, T)
z = [k-1-b.continuity(t0) for t0 in np.unique(b.knots)]
```
Of course, one-line magic gives more compressed code, but often comes at the cost of readability. However this is not too bad, so I'm going to recommend it.